_posts/2020-11-09-beginners-guide-to-citing-open-source-code-in-rmarkdown.md

```diff
@@ -0,0 +1,117 @@
+
+Each BibTeX entry will have a **type**, a **key**, and several **tags** that contain the citation information for the entry. The **key** is used to identify the name of the work you are citing, whereas **tags** represent different types of author information within the entry.
+Here's a BibTeX entry for the `dplyr` package from one of my projects on GitHub:
+
+`@Manual{dplyr,`\
+&ensp;&ensp; `title = {dplyr: A Grammar of Data Manipulation},`\
+&ensp;&ensp; `author = {Hadley Wickham and Romain François and Lionel {`\
+&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp; `Henry} and Kirill Müller},`\
+&ensp;&ensp; `year = {2020},`\
+&ensp;&ensp; `note = {R package version 1.0.2},`\
+&ensp;&ensp; `url = {https://CRAN.R-project.org/package=dplyr},`\
+`}`
+
+Here, the type is **@Manual**, with key **dplyr**. The title, authors, year, version, and url are all **Tags**.
+Note that **Tag names** are not case sensitive.
+You may also notice the **author Tag** goes over the 60 character line limit, so a concatenation is used.
+It would be helpful for you to utilize a series of **@Strings** to abbreviate the names like so:
+
+`@string { hw = "Hadley Wickham" }`\
+`@string { rf = " and Romain Francois" }`\
+`@string { lh = " and Lionel Henry" }`\
+`@string { km = " and Kirill Muller" }`
+
+That way the **author Tag** can look something like this:
+
+`author = {hw # rf # lh # km}`
+
+## Calling Your References
+The **key** in the above example and all of your BibTeX entries is a shorthand name for your reference that acts as the signal to Rmarkdown to search your .bib file for the information to populate your reference in a paragraph.
+The format for that call is `[@key]`.
+You can place the citation call in a paragraph of text easily using square brackets:
+
+"This post was made possible by the information made available on bibtex.org [@bibtex]. Thanks guys!"
+
+It's important to include your reference file in the header of your Rmarkdown file.
+This allows Rmarkdown to search your repository for the .bib file and search it for your references.
+The format is `bibliography: filename.bib` and should go in your **YAML header** adjacent to your title, name and parameters. Here's a typical YAML header for a simple Rmarkdown file:
+
+`---`\
+`title: "Analysis of COVID-era Apple Mobility Data"`\
+`author: "Zachary Houston"`\
+`date: "11/02/2020"`\
+`output: html_document`\
+`bibliography: references.bib`\
+`params:`\
+&ensp;&ensp; `state: "California"`\
+&ensp;&ensp; `data: "data/raw_data/applemobilitytrends-2020-09-24.csv"`\
+&ensp;&ensp; `seqdata: "output/sequence_summary.txt"`\
+`---`
+
+Notice that the entire header is contained by `---` above and below. This defines the header for Rmarkdown.
+
+## Where Do I Find Author Information for Code?
+Depending on what you're citing, it may be as simple as performing a search for the name of the tool or package.
```

Comment:
I'd mention the `citation()` function in R. If you run it with a package name, it gives you the citation info, e.g. `citation("dplyr")`

Reply:
Also you may need to stage things with `git add` after running the above -- I'm not sure off the top of my head.